client/src/components/DropZoneUpload.js

```diff
@@ -1,9 +1,14 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useRef } from "react";
 import { Button } from "@material-ui/core";
 import { DropzoneDialog } from "material-ui-dropzone";
 import axios from "axios";
 
-function DropZoneUpload(props) {
+function DropZoneUpload({
+  uploadLocation,
+  projectId,
+  upload,
+  handleUploadSuccess,
+}) {
```

Comment:
A small question: Is this the preferred method with components?

Reply:
Generally yes, it avoids needing to use `props.projectId`, `props.upload`, etc.